optimum/intel/openvino/modeling_visual_language.py

```diff
@@ -0,0 +1,1192 @@
+                    part_model = model_cls._compile_model(
+                        os.path.join(model_id, part_file_name), kwargs.get("device", "CPU"), kwargs.get("ov_config")
+                    )
+                    kwargs[part] = part_model
+
+        # Load model from hub
+        else:
+            model_file_names = {
+                "language_model": language_model_file_name,
+                "text_embeddings": text_embeddings_file_name,
+                "vision_embeddings": vision_embeddings_file_name,
+            }
+            for part in model_cls.additional_parts:
+                model_file_names[part] = part_file_name = f"openvino_{part}_model.xml"
+
+            file_names = model_file_names.copy()
+            for name, file_name in model_file_names.items():
+                model_cache_path = hf_hub_download(
+                    repo_id=model_id,
+                    filename=file_name,
+                    token=token,
+                    revision=revision,
+                    cache_dir=cache_dir,
+                    force_download=force_download,
+                    local_files_only=local_files_only,
+                )
+                file_names[name] = model_cache_path
+
+            model_save_dir = Path(model_cache_path).parent
+            if not compile_only:
+                language_model = model_cls.load_model(file_names["language_model"], quantization_config)
+                text_embeddings = model_cls.load_model(file_names["text_embeddings"], quantization_config)
+                vision_embeddings = model_cls.load_model(file_names["vision_emnbeddings"], quantization_config)
+                for part in model_cls.additional_parts:
+                    kwargs[part] = model_cls.load_model(file_names[part], quantization_config)
+            else:
+                language_model = model_cls._compile_model(
+                    file_names["language_model"],
+                    kwargs.get("device", "CPU"),
+                    kwargs.get("ov_config"),
+                    model_save_dir,
+                )
+                text_embeddings = model_cls._compile_model(
+                    file_names["text_embeddings"],
+                    kwargs.get("device", "CPU"),
+                    kwargs.get("ov_config"),
+                    model_save_dir,
+                )
+                vision_embeddings = model_cls._compile_model(
+                    file_names["vision_embeddings"],
+                    kwargs.get("device", "CPU"),
+                    kwargs.get("ov_config"),
+                    model_save_dir,
+                )
+                for part in model_cls.additional_parts:
+                    kwargs[part] = model_cls._compile_model(
+                        file_names[part],
+                        kwargs.get("device", "CPU"),
+                        kwargs.get("ov_config"),
+                        model_save_dir,
+                    )
```

Comment:
could the code be simplified with something like 

```suggestion
        model_file_names = {
            "language_model": "openvino_language_model.xml",
            "text_embeddings":  "openvino_text_embeddings_model.xml",
            "vision_embeddings":  "openvino_vision_embeddings_model.xml",
        }
    
        for part in model_cls.additional_parts:
            model_file_names[part] = f"openvino_{part}_model.xml"

        model_cls = MODEL_TYPE_TO_CLS_MAPPING[config.model_type]

        quantization_config = model_cls._prepare_weight_quantization_config(quantization_config, load_in_8bit)
        compile_only = kwargs.get("compile_only", False)

        if os.path.isdir(model_id):
            model_save_dir = Path(model_id)
            file_names = {k:os.path.join(model_id, model_file_names[k]) for k in model_file_names}
        else:
            file_names = model_file_names.copy()
            for name, file_name in model_file_names.items():
                model_cache_path = hf_hub_download(
                    repo_id=model_id,
                    filename=file_name,
                    token=token,
                    revision=revision,
                    cache_dir=cache_dir,
                    force_download=force_download,
                    local_files_only=local_files_only,
                )
                file_names[name] = model_cache_path

            model_save_dir = Path(model_cache_path).parent

        if not compile_only:
            language_model = model_cls.load_model(file_names["language_model"], quantization_config)
            text_embeddings = model_cls.load_model(file_names["text_embeddings"], quantization_config)
            vision_embeddings = model_cls.load_model(file_names["vision_emnbeddings"], quantization_config)
            for part in model_cls.additional_parts:
                kwargs[part] = model_cls.load_model(file_names[part], quantization_config)
        else:
            language_model = model_cls._compile_model(
                file_names["language_model"],
                kwargs.get("device", "CPU"),
                kwargs.get("ov_config"),
                model_save_dir,
            )
            text_embeddings = model_cls._compile_model(
                file_names["text_embeddings"],
                kwargs.get("device", "CPU"),
                kwargs.get("ov_config"),
                model_save_dir,
            )
            vision_embeddings = model_cls._compile_model(
                file_names["vision_embeddings"],
                kwargs.get("device", "CPU"),
                kwargs.get("ov_config"),
                model_save_dir,
            )
            for part in model_cls.additional_parts:
                kwargs[part] = model_cls._compile_model(
                    file_names[part],
                    kwargs.get("device", "CPU"),
                    kwargs.get("ov_config"),
                    model_save_dir,
                )
```

Reply:
simplified